lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -157,9 +159,8 @@ namespace
                 return;
             }
 
-            auto selectionChanged =
-                dynamic_cast<const Ufe::SelectionChanged*>(&notification);
-            if (selectionChanged != nullptr) {
+            if (dynamic_cast<const Ufe::SelectionChanged*>(&notification) ||
+                dynamic_cast<const Ufe::ObjectAdd*>(&notification)) {
```

Comment:
So we listen to `ObjectAdd` for Hydra selection update. This will then make sure a new prim or a variant switched prim to be highlighted if it or any of its ancestor is selected.